chain/mempool.go

```diff
@@ -52,11 +52,19 @@ func (c *cachedInputs) addInput(op wire.OutPoint, txid chainhash.Hash) {
 	// Check if the input already exists.
 	oldTxid, existed := c.inputs[op]
 
+	// If the oldTxid exists and is different from the current txid, we
```

Comment:
Not 100% related to this diff: but can't we just use as integer as the key for the second level of the map? Since given the txid and the index, you already know the input. 

This could allow us to cut down on memory usage a bit, as in _practice_ transaction can't have up to 65k outputs. I think the practical limit is more like 4-5k outputs given current weight constraints.